### PR TITLE
broke some rules

### PR DESCRIPTION
**Pull request by corinnebosley**
Removed deprecated classes from iris.fileformats.rules.py:

Rule
FunctionRule
ProcedureRule
RulesContainer

### CODE REVIEW

**Comment by corinnebosley** [comment]
Closes https://github.com/SciTools/iris/issues/2674
Closes https://github.com/SciTools/iris/issues/2675
Closes https://github.com/SciTools/iris/issues/2676

**Comment by corinnebosley** [comment]
WIP: This PR needs some rethinking and reworking, so I am putting it on hold for now.

**Comment by LukeC92** [comment]
@corinnebosley I see that this Pull Request is addressing #2675 by removing FunctionRule. Several of my Pull Requests have removed small parts of FunctionRule, could these cause conflicts or simplify the issue slightly?

The Pull Requests are  #2791, #2794 and #2796 .

In each case I have removed a smalle deprecated class from lib/iris/fileformats/rules.py and some elif statements in FunctionRule._process_action_result .

Each of the Pull Requests pass all tests locally, so FunctionRule._process_action_result is clearly still able to function without the if statements I remove.

From memory I can remove the deprecated classes without touching FunctionRule at all; it's just there'll be some if statements left in it which will do nothing.

**Comment by corinnebosley** [comment]
@dkillick This morning I have also removed 'Rule' and 'RuleResult' from this module as they were also part of the deprecated rules text-file process.

File: lib/iris/fileformats/rules.py [hunk 1]
```diff
@@ -88,43 +87,6 @@ def as_cube(self):
...
-        _rules_globals.update(iris.coord_systems.__dict__)
-        _rules_globals.update(iris.fileformats.um_cf_map.__dict__)
-        _rules_globals.update(cf_units.__dict__)
-        _import_pending = False
-
-    return _rules_globals.copy()
-
-
 # A flag to control all the text-rules and rules-logging deprecation warnings.
 _enable_rules_deprecations = True
```
Comment (dkillick) [inline lib/iris/fileformats/rules.py]: We'll need to remove this (and the following function) at some point... but not here. How about in https://github.com/SciTools/iris/issues/2670?

**Comment by dkillick** [comment]
This PR had to wait a little to get merged! Good job @corinnebosley

**Pull request merged**